js/tab.js

```diff
@@ -134,15 +134,16 @@ define([
   
   Tab.prototype.detectSyntax = function(userConfig) {
     //settings are async
-    var self = this;
-    Settings.pull("user").then(function(data) {
+    Settings.pull("user").then((data) => {
       var userConfig = data.user;
-      self.setUseSoftTabs(!userConfig.useTabs);
-      self.setTabSize(userConfig.indentation || 2);
-      self.setUseWrapMode(userConfig.wordWrap);
-      self.setWrapLimit(userConfig.wrapLimit || null);
-      self.setNewLineMode(userConfig.lineEnding || "auto");
-      self.setUseWorker(userConfig.useWorker);
+      this.setUseSoftTabs(!userConfig.useTabs);
+      this.setTabSize(userConfig.indentation || 2);
+      this.setUseWrapMode(userConfig.wordWrap);
+      this.setWrapLimit(userConfig.wrapLimit || null);
+      this.setNewLineMode(userConfig.lineEnding || "auto");
+      
+      this.$stopWorker();
```

Comment:
Let me know if you can't reproduce, but in my testing, you don't need to call `$stopWorker` - the `useWorker` config option is a setter that automatically stops and restarts the worker.